### PR TITLE
Move feature column codegen to common path for XGBoost models support

### PR DESCRIPTION
**Pull request by sneaxiy**
Mainly move `generateFeatureColumnCode` to a new file `codegen_feature_column.go`.

### CODE REVIEW

File: pkg/step/feature/derivation.go [hunk 1]
```diff
@@ -133,9 +133,9 @@ func newRowValue(columnTypeList []*sql.ColumnType) ([]interface{}, error) {
 	for idx, ct := range columnTypeList {
 		typeName := ct.DatabaseTypeName()
 		switch unifyDatabaseTypeName(typeName) {
-		case "VARCHAR", "TEXT", "STRING":
+		case "CHAR", "VARCHAR", "TEXT", "STRING":
 			rowData[idx] = new(string)
-		case "INT":
+		case "INT", "TINYINT":
```
Comment (typhoonzero) [inline pkg/step/feature/derivation.go:138]: Will there be some test cases for these data types?

**Review by typhoonzero — DISMISSED** [review]
LGTM generally

File: pkg/sql/codegen/feature/codegen_feature_column.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
```
Comment (typhoonzero) [inline pkg/sql/codegen/feature/codegen_feature_column.go]: Should we move package `feature` under `pkg/step/feature/` or keep it here? Seems we already have a package named `feature` there.

File: pkg/step/feature/derivation.go (hunk 1 above)
Comment (sneaxiy) [reply pkg/step/feature/derivation.go:138]: Done.

File: pkg/sql/codegen/feature/codegen_feature_column.go (hunk 2 above)
Comment (sneaxiy) [reply pkg/sql/codegen/feature/codegen_feature_column.go]: Thanks for recommendation. Moved to `codegen`.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**